app/components/Form/ProjectQuarterlyReportFormSummary.tsx

```diff
@@ -0,0 +1,160 @@
+
+import CUSTOM_DIFF_FIELDS from "lib/theme/CustomDiffFields";
+import { utils } from "@rjsf/core";
+import projectReportingRequirementSchema from "data/jsonSchemaForm/projectReportingRequirementSchema";
+import { ProjectQuarterlyReportFormSummary_projectRevision$key } from "__generated__/ProjectQuarterlyReportFormSummary_projectRevision.graphql";
+import { getFilteredSchema } from "lib/theme/getFilteredSchema";
+import { quarterlyReportUiSchema } from "./ProjectQuarterlyReportForm";
+
+const { fields } = utils.getDefaultRegistry();
+
+interface Props {
+  projectRevision: ProjectQuarterlyReportFormSummary_projectRevision$key;
+  viewOnly?: boolean;
+}
+
+const ProjectQuarterlyReportFormSummary: React.FC<Props> = (props) => {
+  const { summaryProjectQuarterlyReportFormChanges, isFirstRevision } =
+    useFragment(
+      graphql`
+        fragment ProjectQuarterlyReportFormSummary_projectRevision on ProjectRevision {
+          isFirstRevision
+          summaryProjectQuarterlyReportFormChanges: projectQuarterlyReportFormChanges {
+            edges {
+              node {
+                id
+                isPristine
+                newFormData
+                operation
+                formChangeByPreviousFormChangeId {
+                  newFormData
+                }
+              }
+            }
+          }
+        }
+      `,
+      props.projectRevision
+    );
+
+  // Show diff if it is not the first revision and not view only (rendered from the quarterly report page)
+  const renderDiff = !isFirstRevision && !props.viewOnly;
+
+  // If we are showing the diff then we want to see archived records, otherwise filter out the archived quarterly reports
+  let quarterlyReportFormChanges =
+    summaryProjectQuarterlyReportFormChanges.edges;
+  if (!renderDiff)
+    quarterlyReportFormChanges =
+      summaryProjectQuarterlyReportFormChanges.edges.filter(
+        ({ node }) => node.operation !== "ARCHIVE"
+      );
+
+  // Sorting the quarterly report form changes by the reporting requirement index
+  const [sortedQuarterlyReports] = useMemo(() => {
+    const filteredReports = quarterlyReportFormChanges.map(({ node }) => node);
+
+    filteredReports.sort(
+      (a, b) =>
+        a.newFormData.reportingRequirementIndex -
+        b.newFormData.reportingRequirementIndex
+    );
+
+    return [filteredReports];
+  }, [quarterlyReportFormChanges]);
+
+  // Defines if all quarterly reports are pristine
+  const allFormChangesPristine = useMemo(
+    () =>
+      !quarterlyReportFormChanges.some(
+        ({ node }) => node?.isPristine === false || node?.isPristine === null
+      ),
+    [quarterlyReportFormChanges]
+  );
+
+  const quarterlyReportsJSX = useMemo(() => {
+    return sortedQuarterlyReports.map((quarterlyReport, index) => {
+      if (!quarterlyReport) return;
+
+      // Set custom rjsf fields to display diffs
+      const customFields = { ...fields, ...CUSTOM_DIFF_FIELDS };
```

Comment:
`customFields` could be pulled out of the react component altogether, and computed just once at import time (right after line 14)